leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java

```diff
@@ -319,27 +328,6 @@ protected Connector createSecuredConnector(DtlsConnectorConfig dtlsConfig) {
                     localAddress, incompleteConfig.getAddress()));
         }
 
-        // Handle active peers
-        if (incompleteConfig.getMaxConnections() == null)
-            dtlsConfigBuilder.setMaxConnections(coapConfig.getInt(Keys.MAX_ACTIVE_PEERS));
-        if (incompleteConfig.getStaleConnectionThreshold() == null)
-            dtlsConfigBuilder.setStaleConnectionThreshold(coapConfig.getLong(Keys.MAX_PEER_INACTIVITY_PERIOD));
```

Comment:
I put these values in different configuration definitions.
New:

```
CoapConfig.MAX_ACTIVE_PEERS
CoapConfig.MAX_PEER_INACTIVITY_PERIOD
DtlsConfig.DTLS_MAX_CONNECTIONS
DtlsConfig.DTLS_STALE_CONNECTION_THRESHOLD
```

The coap is about tracking MID and messages, dtls about credentials. For me, they are rather soft-related, therefore I separated them.

Reply:
I think it make sense to keep it separated in the config.

But my idea was that it also could make sense to synchronized this when user doesn't change it "manually". 
I feel this kind of default synchronization could make sense to help keeping layers configuration consistent (e.g : https://github.com/eclipse/leshan/pull/1003)

In another hand, maybe this kind of magic could be also surprising for user.

Reply:
> But my idea was that it also could make sense to synchronized this when user doesn't change it "manually".

That will get complex:
Did the user not change the default by the intention to use, what ever the default is? Or because the default value is the one intended?
With that, a "normal" Configuration3.properties will be "empty" and only contains in the comments, what the resulting defaults may be.

For me the "old way" was also in some situation hard to find out, what is effective.
And so I tend to have it "it's as in the Configuration3.properties" without too much magic afterwards.
That includes, that users read the Configuration3.properties "carefully", if something should be adjusted. 
In many cases, that works well. Not in all.